src/runtime_src/core/edge/common/aie_parser.h

```diff
@@ -28,6 +28,12 @@ class device;
 
 namespace edge { namespace aie {
 
+enum class e_module_type {
```

Comment:
I have to ask what's the 'e_' in 'e_module_type'?    

The very nice thing about enum classes is that enumerators must be and are scoped by the class name, so you don't need long names anymore, maybe just 'core', 'dma', 'shim'.